### PR TITLE
OCW News Redesign

### PR DESCRIPTION
**Pull request by ibrahimjaved12**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/1388

#### What's this PR do?
This PR redesigns the OCW News portion of the homepage according to these designs:
[Mobile](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Homepage?page-id=288%3A1298&type=design&node-id=288-1299&viewport=683%2C483%2C0.72&scaling=min-zoom)
[Desktop](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Homepage?page-id=0%3A1&type=design&node-id=44-7&viewport=-277%2C1045%2C0.24&scaling=min-zoom&starting-point-node-id=89%3A2409
)

#### Review Steps
1. Checkout to this branch
2. `yarn start www`

Review will involve checking that the design does not break at any breakpoint. Images, text etc for featured news (the main big news on the left), as well as the news cards in the right side -- their title texts, images, etc.

In line 73, we have:
`{{ range $index, $item := after 1 (first 5 $items) }}
`
If we change `first 5 $items` to something like `first 10 $items`, then we can check for more news cards on the right side -- with different title texts, images of different size -- and we can confirm that they fit well.
And if we change it to something like `first 2 $items` we can also see that the design still works if we don't have enough items for the sidebar to display.

Similarly, in line 60 we have:
`{{ $firstItem := index $items 0 }}
`
For testing purposes, I would make the index 4 because index 4 image has bigger title and to check that truncating was done right.

Similarly, in line 26:
`                    {{ range $index, $newsItem := (first 5 $items) }}`
For mobile carousel view, if we change this to first 10 items we can have more items in the carousel and we can check that they display well.

Check that links redirect correctly to news items and the overall design is as expected.
There were some padding and font size etc related changes that had to be made to improve design after discussion with @mbilalmughal (that may be slightly different from figma designs).

#### Screenshots (if appropriate)
image
image
image

### CODE REVIEW

**Review by HussainTaj-arbisoft — CHANGES_REQUESTED** [review]
Screenshot 2023-06-19 at 2 43 45 PM

I would like to suggest some interactivity improvements.

In the desktop version, the yellow areas are not clickable. Making them clickable is the correct choice in my opinion.

The right-side elements are highlighted when a person hovers over them. However, they are only clickable when the hover is over the title or image. The hover effect and the ability to click should coincide. So, either only the title should trigger the hover effect, or the entire element should be clickable.

@mbilalmughal what do you think?

**Comment by pdpinch** [comment]
Regarding alt text for images, the accessibility manager for Open Learning recommends against using alt text for decorative images, especially if it will be repetitive. 

If we get alt text in the news feed, we can use that. But if there is none, it's probably better to leave it empty, I.e., `alt=””`.

File: base-theme/assets/css/variables.scss [hunk 1]
```diff
@@ -43,6 +43,18 @@ $course-content-link-unclicked: #115f83;
...
 
+// OCW news in homepage
+$featured-news-heading-font-lg: 1.875rem;
+$featured-news-line-height-lg: 2.325rem;
+$featured-news-heading-font-sm: 1.25rem;
+$featured-news-line-height-sm: 1.56rem;
+$featured-news-heading-font-md: 1.4rem;
+$featured-news-line-height-md: 1.8rem;
+$featured-news-date-font: 0.875rem;
+$featured-news-date-line-height: 1rem;
```
[the remark this answers is not in the document]
Comment (ibrahimjaved12) [reply base-theme/assets/css/variables.scss:54]: I think this requires re-evaluating the whole designs, the values used, everything etc. 

The same values may already have a variable, but it's name refers to some other use case -- so for that reason I could not use it. 
And also for the fact that in OCW News Redesign we actually have some different values at different breakpoints. Not all are already here also. 

I think this whole file of variables need a refactoring -- but that only comes with a re-evaluation of the designs and it's scope is out of scope for this PR. Like we need to have some fixed values to follow but we also need design representations that follow these. Only then we can have a thorough refactoring.

File: www/assets/css/home.scss [hunk 2]
```diff
@@ -517,3 +523,198 @@ $promo-carousel-height: 200px;
...
+  #news-carousel-xs-sm {
+    display: none;
+  }
+}
+
+#news-carousel-xs-sm {
+  .carousel-header {
+    margin-bottom: 1.25rem;
+  }
+}
```
[the remark this answers is not in the document]
Comment (ibrahimjaved12) [reply www/assets/css/home.scss:714]: It is used for small screens. Without it you can see we don't have a margin between the buttons and the news image:
image
image

**Comment by ChristopherChudzicki** [comment]
See also https://www.w3.org/WAI/tutorials/images/decorative/

File: www/layouts/partials/ocw_news.html [hunk 3]
```diff
@@ -9,51 +9,93 @@
...
+              <div class="news-card">
+                {{ if $item.image }}
+                <img src="{{$item.image}}" alt="">
+                {{ else }}
+                <img src="/static_shared/images/default_news.png" alt="">
+                {{ end }}
+                <div class="news-card-content">
+                  <div class="news-card-title">
+                    <h3 class="">
+                      {{ $item.title.text | truncate 60 }}
```
Comment (HussainTaj-arbisoft) [inline www/layouts/partials/ocw_news.html:92]: Trim spaces in output
```suggestion
                      {{- $item.title.text | truncate 60 -}}
```

**Review by HussainTaj-arbisoft — CHANGES_REQUESTED** [review]

File: www/layouts/partials/ocw_news.html [hunk 4]
```diff
@@ -9,51 +9,93 @@
...
+                              <div class="featured-news-content">
+                                <div class="featured-news-date">{{ time.Format "January 2, 2006" $newsItem.pubDate.text }}</div>
+                                <h3 class="featured-news-heading mobile">
+                                  <a href="{{ $newsItem.link.text }}" class="text-white text-decoration-none">{{ $newsItem.title.text }}</a>
+                                </h1>
+                              </div>
+                            </div>
+                          </a>
+                        </div>
+                      {{ if (or (eq $modulo (sub $itemsInCarousel 1)) (eq $index (sub 5 1))) }}
```
[the remark this answers is not in the document]
Comment (ibrahimjaved12) [reply www/layouts/partials/ocw_news.html:49]: I had tried getting rid of that equation in some other way but it was causing inconsistencies in the carousel animation. But that makes sense to use same condition for opening/closing tag. Good call getting rid of stale code.

File: www/layouts/partials/ocw_news.html [hunk 5]
```diff
@@ -9,51 +9,93 @@
...
-                          style="background: linear-gradient(to bottom, transparent, #777777), url({{ $newsItem.image }})">
-                      <a href="{{ $newsItem.link.text }}" class="py-2 px-3 text-white text-decoration-none">
-                        {{ $newsItem.title.text }}
-                      </a>
-                    </div>
+              <div class="news-container mobile">
+                <div class="featured-news">
+                  <div class="carousel-inner d-flex px-0">
+                    {{ range $index, $newsItem := (first 5 $items) }}
+                      {{ $modulo := (mod $index $itemsInCarousel) }}
```
[the remark this answers is not in the document]
Comment (ibrahimjaved12) [reply www/layouts/partials/ocw_news.html]: It appears modulo is not actually responsible here for getting back to 1st/Last item in the animation. Hence removed.

**Review by HussainTaj-arbisoft — APPROVED** [review]
LGTM 👍

**Comment by ibrahimjaved12** [comment]
> LGTM 👍

Thanks

**Pull request merged**